wallet-new/src/Cardano/Wallet/API/Request/Sort.hs

```diff
@@ -87,25 +116,20 @@ instance ( Indexable' a
          , IsIndexOf' a ix
          , ToIndex a ix
          , ToSortOperations ixs a
+         , Typeable ix
          )
          => ToSortOperations (ix ': ixs) a where
-  toSortOperations req [] _     =
-      let newOp = SortIdentity
-      in SortOp (newOp :: SortOperation ix a) (toSortOperations req [] (Proxy :: Proxy ixs))
-  toSortOperations req (key:xs) _ =
-      case List.lookup "sort_by" (parseQueryText $ rawQueryString req) of
-          Nothing       ->
-              let newOp = SortIdentity
-              in SortOp (newOp :: SortOperation ix a) (toSortOperations req xs (Proxy @ ixs))
-          Just Nothing  ->
-              let newOp = SortIdentity
-              in SortOp (newOp :: SortOperation ix a) (toSortOperations req xs (Proxy @ ixs))
-          Just (Just v) ->
-              case parseSortOperation (Proxy @a) (Proxy @ix) (key, v) of
-                  Left _      ->
-                      let newOp = SortIdentity
-                      in SortOp (newOp :: SortOperation ix a) (toSortOperations req xs (Proxy @ ixs))
-                  Right newOp -> newOp `SortOp` toSortOperations req xs (Proxy @ ixs)
+    toSortOperations _ [] _     =
+        NoSorts
+    toSortOperations req (key:xs) _ =
+        foldr (either (flip const) SortOp) rest
+            . map (parseSortOperation (Proxy @a) (Proxy @ix) . (,) key)
+            . mapMaybe snd
+            . filter (("sort_by" ==) . fst)
+            . parseQueryText
+            $ rawQueryString req
+      where
+        rest = toSortOperations req xs (Proxy @ ixs)
```

Comment:
The previous implementation had a bug, where multiple `sort_by` in the query would be failed:

```haskell
/foo?sort_by=bar&sort_by=baz
```

The `parseQueryText` would turn that into `[("sort_by", Just "bar"), ("sort_by", Just "baz")]`. When you call `List.lookup`, that would find `"bar"`. If `"bar"` does not match the index, then it is omitted. However, `"bar"` will *always* be what is looked up!

This implementation finds all query parameters that have `"sort_by"` at first, and then attempts to parse all of them into a sort operation. If any succeed, then they are added to the `SortOps`. This allows multiple sorts in a resource.

Reply:
> This allows multiple sorts in a resource.

@parsonsmatt That's neat! I think that sorting over multiple field is a nice property it would be nice to support. Good catch!